hudi-flink-datasource/hudi-flink/src/test/java/org/apache/hudi/sink/TestWriteMergeOnRead.java

```diff
@@ -59,6 +102,113 @@ public void testIndexStateBootstrapWithCompactionScheduled() throws Exception {
     validateIndexLoaded();
   }
 
+  @ParameterizedTest
+  @ValueSource(booleans = {true, false})
+  public void testPrecombinFieldOnSnapshotQuery(boolean enableChangelog) throws Exception {
+    String path = tempFile.getAbsolutePath();
+    EnvironmentSettings settings = EnvironmentSettings.newInstance().inBatchMode().build();
+    TableEnvironment tableEnv = TableEnvironmentImpl.create(settings);
```

Comment:
Is there any possibility to reuse the TestHarness pipeline ?

Reply:
resolved~